cmd/mdatagen/metadata-schema.yaml

```diff
@@ -6,8 +6,8 @@ parent: string
 
 # Required for components (Optional for subcomponents): A high-level view of the development status and use of this component
 status:
-  # Required: The class of the component (For example receiver)
-  class: <receiver|processor|exporter|connector|extension>
+  # Required: The class of the component. receiver_scraper is a receiver which scrapes a fixed set of metrics.
+  class: <receiver|receiver_scraper|processor|exporter|connector|extension>
```

Comment:
This becomes a bit confusing, like it's an entirely different component.

I don't think we should extend this list. The `class` should represent a configuration section. We could add another field like `subclass` or something like that, but I'm not sure why it's needed. The `metrics` section itself can indicate that.

Reply:
> The class should represent a collector configuration section. We could add another field like subclass or something like that

Agreed this would be a better approach 👍 

> I'm not sure why it's needed. The presence of the metrics section itself can indicate that.

This subclass would be required to have the metrics in their `metadata.yaml` (and perhaps also generate functions which use the `scraperhelper` package). 
Just the presence of metrics can indicate that the receiver is a scraper, but I believe we should still call out that metrics are required in the [metadata-schema.yaml](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/cmd/mdatagen/metadata-schema.yaml#L54) for scrapers. I think having the `subclass` field  would make this easier to callout, and we could also add validation that metrics are defined for subclass scraper.

Reply:
I think subclass also makes more sense to me.

> The presence of the metrics section itself can indicate that.

I think it is cleaner to have an explicit flag on the configuration for this. The presence of the metrics section also tells us this is a `receiver`, but I still think we should require `class` to be defined. IMO, the subclass should be source of truth for what kind of component something is, and then we would check for the presence of the `metrics` field.

Reply:
What kind of subclasses do you think of? 

What about prometheus or windowspercounter receiver? They are scraping receivers but don't have predefined set of metrics

Reply:
Apologies for the delay

> What about prometheus or windowspercounter receiver? They are scraping receivers but don't have predefined set of metrics

Right, 'scraper receiver' is not the right name here, I originally discussed this with Mackenzie and suggested this name since all receivers that have a fixed set of metrics are using or can use the scraperhelper, but the important subclass here are the receivers that produce a fixed set of metrics.

> What kind of subclasses do you think of?

Processors that set a fixed set of resource attributes also seem like a relevant class in terms of metadata; I guess there you can argue that the presence of `resource_attributes` would codify that.